rust/arrow/src/array/union.rs

```diff
@@ -698,7 +699,11 @@ mod tests {
             4 * 8 * 4 * mem::size_of::<i32>(),
             union.get_buffer_memory_size()
         );
-        let internals_of_union_array = (8 + 72) + (union.boxed_fields.len() * 144); // Arc<ArrayData> & Vec<ArrayRef> combined.
+        let tagged_pointer_size = POINTER_WIDTH / 4 + POINTER_WIDTH;
+        let internals_of_union_array = tagged_pointer_size
```

Comment:
I struggled to follow the arithmetic here, but perhaps I need to look at the spec again

Reply:
Please do. Calculated nestedness and pointer table from my mind. Just tried to write a little bit more clearly which part of the pointers accounts for which part of the arrays with this pr.